docs/examples/basic-price-oracle/main.go

```diff
@@ -0,0 +1,185 @@
+// This example demonstrates a basic RPC price oracle server that implements the
+// QueryRateTick RPC method. The server listens on localhost:8095 and returns a
+// rate tick for a given transaction type, subject asset, and payment asset. The
+// rate tick is the exchange rate between the subject asset and the payment
+// asset.
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	oraclerpc "github.com/lightninglabs/taproot-assets/taprpc/priceoraclerpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	// serviceListenAddress is the listening address of the service.
+	serviceListenAddress = "localhost:8095"
+)
+
+// RpcPriceOracleServer is a basic example RPC price oracle server.
+type RpcPriceOracleServer struct {
+	oraclerpc.UnimplementedPriceOracleServer
+}
+
+// isSupportedSubjectAsset returns true if the given subject asset is supported
+// by the price oracle, and false otherwise.
+func isSupportedSubjectAsset(subjectAsset *oraclerpc.AssetSpecifier) bool {
+	// Ensure that the subject asset is set.
+	if subjectAsset == nil {
+		return false
+	}
+
+	// In this example we'll only support a single asset.
+	assetIdStr := subjectAsset.GetAssetIdStr()
+	supportedAssetId := "7b4336d33b019df9438e586f83c587ca00fa65602497b9" +
+		"3ace193e9ce53b1a67"
+
+	return assetIdStr == supportedAssetId
+}
+
+// getRateTick returns a rate tick for a given transaction type and subject
+// asset max amount.
+func getRateTick(transactionType oraclerpc.TransactionType,
+	subjectAssetMaxAmount uint64) oraclerpc.RateTick {
+
+	// Determine the rate based on the transaction type.
+	var rate uint64
+	if transactionType == oraclerpc.TransactionType_PURCHASE {
+		// The rate for a purchase transaction is 42,000 asset units per
+		// mSAT.
+		rate = 42_000
+	} else {
+		// The rate for a sale transaction is 40,000 asset units per
+		// mSAT.
+		rate = 40_000
+	}
+
+	// Set the rate expiry to 5 minutes by default.
+	expiry := time.Now().Add(5 * time.Minute).Unix()
+
+	// If the subject asset max amount is greater than 100,000, set the rate
+	// expiry to 1 minute.
+	if subjectAssetMaxAmount > 100_000 {
+		expiry = time.Now().Add(1 * time.Minute).Unix()
+	}
+
+	return oraclerpc.RateTick{
+		Rate:            rate,
+		ExpiryTimestamp: uint64(expiry),
+	}
+}
+
+// QueryRateTick queries the rate tick for a given transaction type, subject
```

Comment:
I think it would be good to further lift this into a high level markdown doc (along with how to set one up, the flags, etc). Don't think we need to block merging this based on that though.

Reply:
I think I've mentioned this to @Liongrass .